src/VectorizeLoops.cpp

```diff
@@ -962,17 +962,23 @@ class VectorSubs : public IRMutator {
             update_replacements();
             // Go over lets which were vectorized and update them according to the current
             // loop level.
-            for (auto it = scope.cbegin(); it != scope.cend(); ++it) {
-                string vectorized_name = get_widened_var_name(it.name());
-                Expr vectorized_value = mutate(it.value());
+            for (auto let = containing_lets.begin(); let != containing_lets.end(); let++) {
```

Comment:
Do we need to add comment(s) here and below about why we are doing this?

Reply:
I updated the comments, but it really was just my mistake incorrectly assuming that Scope preserves insertion order (which it does but only within *specific* name).